src/dialect.js

```diff
@@ -892,19 +897,16 @@ class Dialect {
    * @return String            The SQL meta string.
    */
   constraint(name, constraint, options) {
-    constraint = extend({}, { options: [] }, constraint);
-
-    var meta = this._constraints[name];
-    var template = meta ? meta.template : undefined;
+    const _constraint = Object.assign({}, { options: [] }, constraint);
+    const meta = this._constraints[name];
+    const template = meta ? meta.template : undefined;
     if (!template) {
       throw new Error("Invalid constraint template `'" + name + "'`.");
     }
 
-    var data = {};
-
-    for (var name in constraint) {
-      var value = constraint[name];
+    const data = Object.create(null);
```

Comment:
this scopes data to the for loop, looks like it access out side of the loop? https://github.com/helpdotcom/sql-dialect/pull/1/files#diff-55d975ba07a03aa0a8f2a8e9b78546d8R940

Reply:
data is outside the loop

Reply:
Ugh... yes it is... sorry